src/webargs/core.py

```diff
@@ -267,6 +308,61 @@ def _get_schema(self, argmap: ArgMap, req: Request) -> ma.Schema:
             raise TypeError(f"argmap was of unexpected type {type(argmap)}")
         return schema
 
+    def _prepare_for_parse(
+        self,
+        argmap: ArgMap,
+        req: Request | None = None,
+        location: str | None = None,
+        unknown: str | None = _UNKNOWN_DEFAULT_PARAM,
+        validate: ValidateArg = None,
+    ) -> tuple[None, Request, str, CallableList, ma.Schema]:
+        # validate parse() arguments and handle defaults
+        # (shared between sync and async variants)
+        req = req if req is not None else self.get_default_request()
+        if req is None:
+            raise ValueError("Must pass req object")
+        location = location or self.location
+        validators = _ensure_list_of_callables(validate)
+        schema = self._get_schema(argmap, req)
+        return (None, req, location, validators, schema)
+
+    def _process_location_data(
+        self,
+        location_data: typing.Any,
+        schema: ma.Schema,
+        req: Request,
+        location: str,
+        unknown: str | None,
+        validators: CallableList,
+    ):
+        # after the data has been fetched from a registered location,
+        # this is how it is processed
+        # (shared between sync and async variants)
+
+        # when the desired location is empty (no data), provide an empty
+        # dict as the default so that optional arguments in a location
+        # (e.g. optional JSON body) work smoothly
+        if location_data is missing:
+            location_data = {}
+
+        # precedence order: explicit, instance setting, default per location
+        unknown = (
+            unknown
+            if unknown != _UNKNOWN_DEFAULT_PARAM
+            else (
+                self.unknown
+                if self.unknown != _UNKNOWN_DEFAULT_PARAM
+                else self.DEFAULT_UNKNOWN_BY_LOCATION.get(location)
+            )
+        )
+        load_kwargs: dict[str, typing.Any] = {"unknown": unknown} if unknown else {}
+        preprocessed_data = self.pre_load(
+            location_data, schema=schema, req=req, location=location
+        )
+        data = schema.load(preprocessed_data, **load_kwargs)
+        self._validate_arguments(data, validators)
```

Comment:
Right now, several of these methods are still synchronous-only. But if we take this model further, we could allow
```python
class MyParser(DjangoParser):
    async def pre_load(...): ...
```
and properly detect and `await` that in here.

It would bulk up the amount of code we'd have to duplicate, but with some nice structure in place for it, we might not find it too difficult. It's not essential and not supported today, even by the AsyncParser, so I don't see it as necessary.